src/Settings.php

```diff
@@ -680,4 +703,36 @@ public function remove_notices() { // phpcs:ignore Generic.Metrics.NestingLevel.
 			}
 		}
 	}
+
+	/**
+	 * Check the sample file.
+	 *
+	 * @since 1.4.0
+	 */
+	public function check_sample() {
+
+		// verify file_upload_types_nonce nonce.
+		check_ajax_referer( 'file_upload_types_nonce', 'nonce' );
+
+		// @todo: add nonce verification.
+		// phpcs:disable phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.NonceVerification.Missing
+		if ( isset( $_FILES['file'] ) ) {
+			$sample         = $_FILES['file'];
+			$sample['name'] = preg_replace( '/[^a-zA-Z0-9\._-]/', '', $sample['name'] );
+			$sample['name'] = uniqid() . '_' . $sample['name'];
+
+			$finfo     = finfo_open( FILEINFO_MIME_TYPE );
+			$mime_type = finfo_file( $finfo, $sample['tmp_name'] );
+			$extension = pathinfo( $sample['name'], PATHINFO_EXTENSION );
+
+			finfo_close( $finfo );
```

Comment:
### Extension.

1. The regex we can replace with the `sanitize_file_name` function. No regex no problem.
2. I didn't get why we need `uniqid` here. If we really need it, please add a comment since it's not obvious.

### Mime type

I read all the threads in issue and Slack and got the mime-type hell.

But according to PHP documentation, we can replace `finfo...` functions with `mime_content_type`. It looks like under the hood it's the same, but it's much easier to read.

Reply:
Thanks for the suggestions 1. and 2. I changed that (about uniqid, I frankly don't remember why I used  this, I wrote some parts of this code 1,5 year ago, now I removed it). 

About `finfo` I will leave it as is. I want to have this code checking the type the same way as build in Wordpress function `wp_check_filetype_and_ext` does because we need to have the same results when user start uploading real files. For now both `finfo` and `mime_content_type` seems to work the same. But I prefer to use the same WP is using